src/coresyftools/coresyftools/gpt_tool.py

```diff
@@ -0,0 +1,101 @@
+'''Defines a CoReSyF Tool consisting of a SNAP single gpt operation.'''
+import os
+import subprocess
+import re
+from tool import CoReSyFTool
+from manifest import InvalidManifestException
+
+class TooManyInputArgumentsException(Exception):
+    pass
+
+class ToManyOutputArgumentsException(Exception):
+    pass
+
+class GPTExecutionException(Exception):
+    '''Error occurred during a SNAP gpt execution'''
+
+    ERROR_REGEX = re.compile('Error:')
+
+    def __init__(self, returncode, msg):
+        self.returncode = returncode
+        self.errors = [line[7:] for line in msg.split(
+            os.linesep) if self.ERROR_REGEX.match(line)]
+        super(GPTExecutionException, self).__init__(self.errors)
+
+
+class GPTGraphFileNotFound(Exception):
+    pass
+
+
+class GPTCoReSyFTool(CoReSyFTool):
+    '''CoReSyF Tool consisting of a single SNAP gpt operation.'''
+
+    DEFAULT_FORMAT = 'GeoTIFF-BigTIFF'
+    DEFAULT_EXT = 'tif'
+    DEFAULT_GPT_GRAPH_FILE_NAME = 'gpt_graph.xml'
+
+    def _validate_operation(self, operation):
+        if 'operation' not in operation and ('graph' not in operation or not operation['graph']):
+            raise InvalidManifestException('A operation or graph flag should be present.')
+        if 'operation' in operation and 'graph' in operation and operation['graph']:
+            raise InvalidManifestException('Can not be operation and graph at same time.')
+        if 'graph' in self.operation and self.operation['graph']:
+            graph_file = os.path.join(self.context_directory, self.DEFAULT_GPT_GRAPH_FILE_NAME)
+            if not os.path.exists(graph_file):
+                raise GPTGraphFileNotFound(graph_file)
+
+    def run(self, bindings):
+        if len(self.arg_parser.inputs) > 1:
+            raise TooManyInputArgumentsException()
+        if len(self.arg_parser.outputs) > 1:
+            raise TooManyInputArgumentsException()
+
+        bindings = bindings.copy()
+        operator = None
```

Comment:
Here would be a good place to invoke the `validate_operation`method.

Reply:
You almost cached me. But my test ( `test_gpt_tool::TestGPTTool::test_missing_operation_and_graph` ), saved me. Such method is already being called by the the super-class, `CoReSyFTool` `__init__`. The method definition in this class is an override. Anyway, even I forgot that. So, I will signal that. I don't know if there is a recommended way to do it, I am just putting a commend.